scripts/update_zenodo.py

```diff
@@ -0,0 +1,180 @@
+        while chunk := file.read(4096):
+            md5_hash.update(chunk)
+    return md5_hash.hexdigest()
+
+
+# remove draft
+def rm_draft(ACCESS_TOKEN, new_deposition_id):
+    r = requests.delete(f'https://zenodo.org/api/deposit/depositions/{new_deposition_id}',
+        params={'access_token': ACCESS_TOKEN})
+    if r.status_code == 204:
+        logger.info(f"Draft discarded")
+    else:
+        logger.info(f"Failed to discard draft, error code: {r.status_code}")
+
+
+if __name__ == "__main__":
+    if not (os.getcwd() == REPO_ROOT):
+        sys.exit(
+            "ALL SCRIPTS MUST BE RUN FROM THE REPO ROOT!!\n"
+            f"\tCurrent directory: {os.getcwd()}\n"
+            f"\tRepo root        : {REPO_ROOT}\n"
+        )
+
+    script_name = os.path.basename(__file__)
+    logger = get_logger(LOG_DIR, LOG_FNAME, script_name=script_name, also_print=True)
+    logger.info("-" * 50)
+    logger.info(f"Begin script: {__file__}")
+
+    # Get list of depositions
+    r = requests.get('https://zenodo.org/api/deposit/depositions',
+                    params={'access_token': ACCESS_TOKEN})
+    if r.status_code == 200:
+        logger.info(f"Retrieved all deposition info")
+    else:
+        logger.info(f"Failed to retrieve the list of depositions, error code: {r.status_code}")
+        try:
+            logger.info(f"trace: {r.json()}")
+        except:
+            logger.info("no trace available")
+        finally:
+            exit()
+    all_depo = r.json()
+
+    # find the corresponding depo
+    deposition_id = -1
+    for depo in all_depo:
+        if depo["conceptrecid"] == conceptrecid:
+            deposition_id = depo['links']['latest'].split('/')[-1]
+            break
+    if deposition_id == -1: logger.info(f"Failed to match the given conceptrecid: {conceptrecid}"); exit()
+
+    # create a new version from the previous version as draft
+    url_new_version = f"https://zenodo.org/api/deposit/depositions/{deposition_id}/actions/newversion"
+    response = requests.post(url_new_version, params={'access_token': ACCESS_TOKEN})
+    json_response = response.json()
+
+    if response.status_code != 201:
+        logger.info(f"Failed to create draft. Status code: {response.status_code}")
+        logger.info(json_response)
+    else:
+        new_deposition_id = json_response['links']['latest_draft'].split("/")[-1]
+        logger.info("Draft created.")
+        logger.info(f"New deposition ID: {new_deposition_id}")
+
+        # Retrieve list of local files
+        file_names = [f for f in os.listdir(folder_path) if os.path.isfile(os.path.join(folder_path, f)) and not f.startswith('.')]
+        remote_files = {x['filename']:(x['checksum'],x['id']) for x in json_response["files"]}
+        to_be_updated = []
+        # iterate through local files and either update or upload
+        for filename in file_names:
+            file_path = f"{folder_path}/{filename}"
+            if filename in remote_files:
```

Comment:
Why do we need this block? Old files should not be changed/updated, I don't think. Is that right? I am worried that something weird will happen and we'll accidentally delete all of the old files... Can we just skip files that we already have in the remote set? LMK what you think.

Reply:
You are right, in our case the old files shouldn't change by any means. I'll skip these files instead.